taichi/backends/device.h

```diff
@@ -498,8 +514,9 @@ class Surface {
   virtual ~Surface() {
   }
 
-  virtual DeviceAllocation get_target_image() = 0;
-  virtual void present_image() = 0;
+  virtual std::pair<DeviceAllocation, StreamSemaphore> get_target_image() = 0;
```

Comment:
not: Better create a small struct instead of std::pair, otherwise it's arcane to future readers when there are many first, second...